src/collective/fingerpointing/upgrades/configure.zcml

```diff
@@ -0,0 +1,29 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:five="http://namespaces.zope.org/five"
+    xmlns:i18n="http://namespaces.zope.org/i18n"
+    xmlns:gs="http://namespaces.zope.org/genericsetup">
+
+  <include file="profiles.zcml" />
+
+  <gs:registerProfile
```

Comment:
we have been doing this for a while in packages like collective.cover with very good result, but is not that complex as you expect.

check: https://github.com/collective/collective.cover/blob/1.0a12/src/collective/cover/tests/test_upgrades.py

I will work in this PR in the afternoon to help on those things and to test it.